src/CG/CG_elem.cpp

```diff
@@ -2499,6 +2497,25 @@ bool intersectBoxSphere( array3D const &A0, array3D const &A1, array3D const &ce
 //to levelset
 //to levelset }
 
+/*!
+ * checks if points lies on line defined by two given points in 3D
+ * \param[in] P point coordinates
+ * \param[in] P1 point belonging to the line
+ * \param[in] P2 point belonging to the line
+ * \param[in] distanceTolerance if distance among features exceed this value they are considered as not intersecting
+ * \return if point lies on segment
+ */
+bool intersectPointLine( array3D const &P, array3D const &P1, array3D const &P2, const double distanceTolerance)
```

Comment:
The functions that operates on lines should take in input a point on the line and the direction of the line (see other *Line* functions):
```
/*!
 * checks if points lies on line in 3D
 * \param[in] P point coordinates
 * \param[in] Q point on line
 * \param[in] n line direction
 * \param[in] distanceTolerance if distance among features exceed this value they are considered as not intersecting
 * \return if point lies on line
 */
bool intersectPointLine( array3D const &P, const array3D &Q, const array3D &n, const double distanceTolerance)
{
    array3D xP;
    if(distancePointLine(P, Q, n, xP) > distanceTolerance){
        return false;
    }

    return true;
}
```

Reply:
Done!